parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -102,8 +109,11 @@ def __init__(self,
         self.uid = uid
 
         cores_on_node = multiprocessing.cpu_count()
+        available_mem_on_node = round(psutil.virtual_memory().available / (2**30), 1)
+
         self.max_workers = max_workers
         self.worker_count = min(max_workers,
+                                math.floor(available_mem_on_node / mem_per_worker),
```

Comment:
I think this would raise exception if `mem_per_worker` is not set (default to 0)